src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/comment/services/LemmyCommentReplyService.java

```diff
@@ -46,40 +56,52 @@ private CommentReplyView.CommentReplyViewBuilder commentReplyViewBuilder(
     final Comment lemmyComment = conversionService.convert(commentReply.getComment(),
         Comment.class);
 
-    final CommentAggregates commentAggregates = conversionService.convert(
-        commentReply.getComment().getCommentAggregate(), CommentAggregates.class);
+    final CommentAggregates commentAggregates = conversionService.convert(commentReply.getComment()
+        .getCommentAggregate(), CommentAggregates.class);
 
-    final Post lemmyPost = conversionService.convert(commentReply.getComment().getPost(),
-        Post.class);
+    final Post lemmyPost = conversionService.convert(commentReply.getComment()
+        .getPost(), Post.class);
 
-    final Community lemmyCommunity = conversionService.convert(
-        commentReply.getComment().getCommunity(), Community.class);
+    final Community lemmyCommunity = conversionService.convert(commentReply.getComment()
+        .getCommunity(), Community.class);
 
-    final int personVote = commentLikeService.getPersonCommentVote(Person,
+    final int personVote = commentLikeService.getPersonCommentVote(person,
         commentReply.getComment());
 
+    final boolean creatorIsAdmin = RolePermissionService.isAdmin(creator);
+    boolean creatorIsModerator = false;
+    boolean creatorBannedFromCommunity = false;
+    boolean creatorIsBlocked = false;
+    if (!creatorIsAdmin) {
+      creatorIsBlocked = linkPersonPersonService.hasLink(creator, person,
+          LinkPersonPersonType.blocked);
+      creatorBannedFromCommunity = linkPersonCommunityService.hasLink(creator,
+          commentReply.getComment()
+              .getCommunity(), LinkPersonCommunityType.banned);
+    } else {
+      creatorIsModerator = linkPersonCommunityService.hasAnyLink(creator, commentReply.getComment()
```

Comment:
If the creator is an admin, why do we care if they are also a moderator? I think the logic flow should be something like:

If is not banned check blocked
If is not blocked, skip all other checks.
If is blocked check if moderator
If is not moderator check if admin

Since the majority of users that will pass through this flow will not be admins or mods, it's probably more efficient to check this way. If the user is neither blocked nor banned, do we need to know if they are a mod or admin? Since we know an admin and mod cannot be banned, if the user is banned we don't need to check those. If a user is blocked, we still need to check admin/mod because maybe you block a user in one community, but they are a mod in another. Then your block would work for the one community, but not the one they are a mod in. Make sense? (Or am I missing something?). And if the user is an admin, tough. You can't block them. Report them to another admin or move to another instance.